static/js/jira-connect-ghe.js

```diff
@@ -0,0 +1,65 @@
+const ALLOWED_PROTOCOLS = ["http:", "https:"];
+const GITHUB_CLOUD = ["github.com", "www.github.com"];
+
+/**
+ * Method that checks the validity of the passed URL
+ *
+ * @param {string} inputURL
+ * @returns {boolean}
+ */
+const checkValidGHEUrl = inputURL => {
+	try {
+		const { protocol, hostname } = new URL(inputURL);
+
+		if (!ALLOWED_PROTOCOLS.includes(protocol)) {
+			return false;
+		}
+		// This checks whether the hostname whether there is an extension like `.com`, `.net` etc.
+		if (hostname.split('.').length < 2) {
+			return false;
+		}
+		if (GITHUB_CLOUD.includes(hostname)) {
+			// TODO: Need to alert the users that this URL is GitHub cloud and not Enterprise
+			// Waiting for design: https://softwareteams.atlassian.net/browse/ARC-1418
+			return false;
```

Comment:
I don't think we should return false here otherwise it will render the invalid URL message. I'd say, until we have the design update with how we should render the 'Were you meaning to connect to cloud?' message, you just drop a console.log in here. Something like `console.log('Cloud URL entered.');` just so it's clear this is not an error state.

Reply:
Agreed. 
This method `checkValidGHEUrl` was build to check if the URL is valid or not and this GitHub cloud check is unrelated to that. 
So, I'm separating this into a new method which only checks if the passed valid URL is a GitHub cloud or enterprise.